src/korma/core.clj

```diff
@@ -743,25 +767,36 @@
   (let [sub-ent (assoc-db-to-entity query sub-ent)
         [ent-key sub-ent-key] (get-join-keys rel (:ent query) sub-ent)]
     (post-query query
-                (partial map
-                         (fn [ent]
-                           (merge-with-unique-keys (get-key-naming-strategy query)
-                                                   ent
-                                                   (first
-                                                     (select sub-ent
-                                                             (body-fn)
-                                                             (where {sub-ent-key (get ent ent-key)})))))))))
+      (partial map
+         (fn [ent]
+           (merge-with-unique-keys (get-key-naming-strategy query)
+              ent
+              (first
+                (select sub-ent
+                  (body-fn)
+                  (where
+                    (if (seq? ent-key)
```

Comment:
implemented composite key support here by handling seq keys differently; white-space changes are also intentional to reduce line length and make the code more readable.